Qute docs improvement (#44289)
quarkusio/quarkus: docs/src/main/asciidoc/qute.adoc

There's an alternate way to declare your templates in your Java code, which relies on the following convention:

- Organise your template files in the `/src/main/resources/templates` directory, by grouping them into one directory per resource class. So, if

>   your `HelloResource` class references two templates `hello` and `goodbye`, place them at `/src/main/resources/templates/ItemResource/hello.txt`

Reviewer: ```suggestion
  your `HelloResource` class references two templates `hello` and `goodbye`, place them at `/src/main/resources/templates/HelloResource/hello.txt`
```

----
<1> `{name}` is a value expression that is evaluated when the template is rendered.

.HelloResource/goodebye.txt
[source]
----
Goodbye {name}! <1>
----
<1> `{name}` is a value expression that is evaluated when the template is rendered.

> 

Reviewer: These lines should not be added, imho.
But it is important to rename [line 211](https://github.com/quarkusio/quarkus/blob/de0a859622edba0830e2427ba182f2cf2103d2f5/docs/src/main/asciidoc/qute.adoc?plain=1#L211) from `HelloResource.java` to `GoodbyeResource.java`: https://github.com/quarkusio/quarkus/blob/de0a859622edba0830e2427ba182f2cf2103d2f5/docs/src/main/asciidoc/qute.adoc?plain=1#L211-L226

Reviewer: Also, add
```java
public static native TemplateInstance goodbye(String name);
```
after https://github.com/quarkusio/quarkus/blob/de0a859622edba0830e2427ba182f2cf2103d2f5/docs/src/main/asciidoc/qute.adoc?plain=1#L194
<br>
And change https://github.com/quarkusio/quarkus/blob/de0a859622edba0830e2427ba182f2cf2103d2f5/docs/src/main/asciidoc/qute.adoc?plain=1#L231 in GoodbyeResource.java to
```java
return HelloResource.Templates.goodbye(name);
```

Reviewer: > These lines should not be added, imho.

Sorry, I was confused. They should stay of course.
But there's a typo in goodbye.txt: https://github.com/quarkusio/quarkus/blob/f32455e7570b08a41ace95fafa3f9223e6404104/docs/src/main/asciidoc/qute.adoc?plain=1#L165

And https://github.com/quarkusio/quarkus/blob/f32455e7570b08a41ace95fafa3f9223e6404104/docs/src/main/asciidoc/qute.adoc?plain=1#L156 should be changed to
```md
We'll start with two very simple templates:
```

> Now let's declare and use those templates in the resource class.

Reviewer: ```suggestion
Now let's declare and use this template in the resource class.
```

    @CheckedTemplate
    public static class Templates {
        public static native TemplateInstance hello(String name); <1>
        public static native TemplateInstance goodbye(String name); <2>
    }

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public TemplateInstance get(@QueryParam("name") String name) {
        return Templates.hello(name); <3>
    }
}
----
<1> This declares a template with path `templates/HelloResource/hello`.
<2> This declares a template with path `templates/HelloResource/goodbye`.
<3> `Templates.hello()` returns a new template instance that is returned from the resource method. Note that we don't trigger the rendering - this is done automatically by a special `ContainerResponseFilter` implementation.

NOTE: Once you have declared a `@CheckedTemplate` class, we will check that all its methods point to existing templates, so if you try to use a template from your Java code and you forgot to add it, we will let you know at build time :)

Keep in mind this style of declaration allows you to reference templates declared in other resources too:

> .GoodbyeResource.java

Reviewer: Also change
```java
@Path("goodbye")
public class GoodbyeResource {
```
to
```java
@Path("greeting")
public class GreetingResource {
```

> .GoodbyeResource.java

Reviewer: Don't forget to include [above changes](https://github.com/quarkusio/quarkus/pull/44289#discussion_r1850587033).

Keep in mind this style of declaration allows you to reference templates declared in other resources too:

> .Goodbye.java

Reviewer: ```suggestion
.GoodbyeResource.java
```

@CheckedTemplate
public class Templates {
    public static native TemplateInstance hello(String name); <1>

>     public static native TemplateInstance goodbye(String name); <2>

Reviewer: Good catch, but this line needs also to be added in here:
https://github.com/quarkusio/quarkus/blob/f8e042f2caedeee44818d706980d4bc078e5c4e6/docs/src/main/asciidoc/qute.adoc?plain=1#L192-L195
